server/mscalendar/availability.go

```diff
@@ -35,11 +35,25 @@ func (m *mscalendar) SyncStatusAll() (string, error) {
 		return "", err
 	}
 
-	mmIDs := userIndex.GetMattermostUserIDs()
+	index := userIndex.GetMattermostUserIDs()
+	mmIDs := []string{}
+	for _, id := range index {
+		if id != m.Config.BotUserID {
```

Comment:
Is this the best place to do filtering? Maybe at the user-API level is better? Also, why would the Bot user ID be in the list to start with, out of curiosity?

Reply:
The bot is inside of the index because it is placed there when the oauth completes. We can/should avoid placing the bot's account in the index.

Reply:
Right. I like the precaution - letting it through would be wasteful if harmless.

Reply:
@levb

> Maybe at the user-API level is better?

I don't see how that is an option, as this is for the status sync job

>  We can/should avoid placing the bot's account in the index.

The `UserStore` does not have access to the config (which i agree with that decision), but this means it does not know what the bot's user id is. I'm thinking the store should have a separate method for storing a user in the index, but then the caller of `StoreUser()` will always need to know to call `StoreUserIndex()` as well.